File: .guides/tests/ch-6.js

collisionEvent = null;
var _count = 0;
var _last = '';
var _true = true;

var fileName = window.location.origin + '/public/py/' + window.testEnv.cmd + '.py';

$.get(fileName)
.success(function(data) {

  if(data == 404) {
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, fileName + ' not found'); 
  }

  try {        
    createEmptyMaze();
    
    // load python
    var module = Sk.importMainWithBody("", false, data);

    // get reference to functions

    collisionEvent = module.tp$getattr('collisionEvent');
    
    // test

    if(typeof collisionEvent != 'undefined') {

      window.showMessage = function(val) {
        _count++;
        _true = _true && (_last != val);
        _last = val;
      }

      Sk.misceval.callsim(collisionEvent, new Sk.builtin.str('wall'));
      Sk.misceval.callsim(collisionEvent, new Sk.builtin.str('monster'));
      Sk.misceval.callsim(collisionEvent, new Sk.builtin.str('energy'));
      Sk.misceval.callsim(collisionEvent, new Sk.builtin.str('goal'));

      if(_count != 4 || !_true) {
        return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
      }   
    }
    else {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }

    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');    

  }
  catch(exception) {
    console.log('skulpt exception');
    console.log(exception);
    codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.toString()); 
  }
})
.fail(function(jqxhr, settings, exception) {
  console.log('jqxhr fail');
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});    

File: .guides/tests/ch-6.py

message = ''

def showMessage(val):
  global message
  message = val

try:
  execfile('/home/codio/workspace/public/py/ch-6.py')
  
  collisionEvent('wall')
  if not 'wall' in message.lower():
    raise ValueError("collisionEvent is not handling the 'wall' type correctly")
  
  collisionEvent('monster')
  if not 'monster' in message.lower():
    raise ValueError("collisionEvent is not handling the 'monster' type correctly")
  
  collisionEvent('energy')
  if not 'energy' in message.lower():
    raise ValueError("collisionEvent is not handling the 'energy' type correctly")
  
  collisionEvent('goal')
  if not ('goal' in message.lower() or 'exit' in message.lower()):
    raise ValueError("collisionEvent is not handling the 'goal' type correctly")
      
  print 'well done'
  exit(0)
  
except (IOError, SyntaxError, ValueError) as e:
  print str(e)
  exit(1)
  
print 'Not quite right, try again!'
exit(1)
